### PR TITLE
Fixed typo in method declaration

### PR DESCRIPTION
**Pull request by ccasciotti**
Fixed a typo in method declaration, changed `proccessAdditionalValidation` in `processAdditionalValidation`

### CODE REVIEW

**Comment by adragus-inviqa** [comment]
Please don't change the API of `AbstractCarrierInterface`, as much as I want it, too.

_Add_ a new method, deprecate the old one, and delegate old to new.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]
Hi,
Unfortunately we can not accept this PR in it's current state due to the fact, that backwards compatibility policy prohibits renaming of methods. 
Please refer to the comment https://github.com/magento/magento2/pull/9154#issuecomment-292206386 if you really wish to implement the fix.

**Comment by ccasciotti** [comment]
I've modified PR, please let me know if now it's ok. I have a question, in a future minor version (ex. 2.2) the API will be editable or it has to remain unchangeable?

**Comment by adragus-inviqa** [comment]
@ccasciotti - thanks for the change. One more nit, if I may: can you tell others what they should be using instead? Something like `@deprecated Use processAdditionalValidation() instead`. It really helps.

Also, removing an interface method is a BC break, as other potential 3rd-party implementations will fail. And BC-breaks should be only made in major versions, not minor. Speaking of which, I don't think M2 considers the `x` in `2.x` as the _minor_ version. They don't do semver (otherwise, the next major version would be M3), and they don't have to. I'm not sure, but `x` in this case is considered the major version by M2. 

I still don't know when they'll clean up all the deprecated code, though.

File: app/code/Magento/Shipping/Model/Carrier/AbstractCarrier.php [hunk 1]
```diff
@@ -325,6 +325,7 @@ public function checkAvailableShipCountries(\Magento\Framework\DataObject $reque
      *
      * @param \Magento\Framework\DataObject $request
      * @return $this|bool|\Magento\Framework\DataObject
+     * @deprecated
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function proccessAdditionalValidation(\Magento\Framework\DataObject $request)
```
Comment (adragus-inviqa) [inline app/code/Magento/Shipping/Model/Carrier/AbstractCarrier.php]: This method's phpdoc should just be `{@inheritdoc}`, precisely to avoid duplication.

Comment (ccasciotti) [reply app/code/Magento/Shipping/Model/Carrier/AbstractCarrier.php]: I agree with you, change done.

**Comment by ccasciotti** [comment]
@adragus-inviqa i've added the explanation on deprecated method.
Thanks for explanation of magento's major/minor version 

@ishakhsuvarov i've implemented requested changes

**Comment by okorshenko** [comment]
Hi @ccasciotti 
Thank you for your contribution. Please, take a look at http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/ 
According to our policy, Introduction of a method to a class or interface is a [prohibited code change](http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/#introduction-of-a-method-to-a-class-or-interface).

Unfortunately, we can not modify existing @api interfaces.

**Comment by okorshenko** [comment]
Hi @ccasciotti 
`\Magento\Shipping\Model\Carrier\AbstractCarrierInterface` and `\Magento\Shipping\Model\Carrier\AbstractCarrier` is a legacy implementation and should be deprecated in favor of `\Magento\Shipping\Model\Carrier\CarrierInterface`.
Unfortunatelly, we can not deprecate these interfaces right now because in order to implement shipping integration requires extending from `AbstractCarrier`.

So we may accept fix in abstract `AbstractCarrier` class but not in `AbstractCarrierInterface` and this won't be BiC. But there are no lots of benefits from such code change.

Thank you for your contribution. But we need to reject this PR.

**Pull request closed without merging**